### PR TITLE
refactor: Inject storage EVERYWHERE [part 4]

### PR DESCRIPTION
**Pull request by joaquim-verges**
This update restructures the exports and re-implements various wallet and transaction hooks. Key changes include moving files to new locations, renaming certain modules, and updating functionalities to streamline the overall structure. These changes aim to enhance clarity, maintainability, and the extensibility of the codebase.

---

---

## PR-Codex overview
The focus of this PR is to refactor wallet-related hooks and storage, introduce connection manager, and update import paths.

### Detailed summary
- Refactored wallet-related hooks and storage
- Introduced `ConnectionManager`
- Updated import paths for storage and hooks
- Added new hooks for wallet connection and status
- Updated `AsyncStorage` import path
- Removed unused hooks and storage files

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/TokenSelector.tsx`, `packages/thirdweb/src/react/native/hooks/wallets/useConnectedWallets.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useActiveAccount.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useActiveAccount.ts`, `packages/thirdweb/src/react/web/ui/TransactionButton/TransactionButton.test.tsx`, `packages/thirdweb/src/react/core/connectionManager.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/NetworkSelector.tsx`, `packages/thirdweb/src/react/web/hooks/wallets/useIsAutoConnecting.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useIsAutoConnecting.ts`, `packages/thirdweb/src/react/core/hooks/transaction/useSimulateTransaction.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/fiat/PostOnRampSwap.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/PayWithCrypto.tsx`, `packages/thirdweb/src/react/web/hooks/wallets/useActiveWalletChain.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useActiveWalletChain.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/main/useUISelectionStates.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/BuyScreen.tsx`, `packages/thirdweb/src/react/web/hooks/wallets/useSetActiveWallet.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useSetActiveWallet.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/SendFunds.tsx`, `packages/thirdweb/src/react/core/hooks/transaction/useEstimateGas.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/ViewFunds.tsx`, `packages/thirdweb/src/react/web/wallets/in-app/storage.ts`, `packages/thirdweb/src/react/core/hooks/transaction/useEstimateGasCost.ts`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletSocialLogin.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/tx-history/BuyTxHistory.tsx`, `packages/thirdweb/src/react/web/hooks/wallets/useDisconnect.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useDisconnect.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useSetActiveWalletConnectionStatus.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useSetActiveWalletConnectionStatus.ts`, `packages/thirdweb/src/wallets/manager/index.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useActiveWalletConnectionStatus.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useActiveWalletConnectionStatus.ts`, `packages/thirdweb/src/react/core/hooks/auth/useSiweAuth.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useSwitchActiveWalletChain.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useSwitchActiveWalletChain.ts`, `packages/thirdweb/src/react/core/hooks/transaction/useSendAndConfirmTransaction.ts`, `packages/thirdweb/src/react/core/hooks/transaction/useSendBatchTransaction.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useAutoConnect.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useAutoConnect.ts`, `packages/thirdweb/src/react/web/ui/hooks/useSendToken.ts`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletPassKeyLogin.tsx`, `packages/thirdweb/src/react/web/wallets/in-app/InAppWalletOTPLoginUI.tsx`, `packages/thirdweb/src/wallets/wallet-connect/controller.ts`, `packages/thirdweb/src/react/web/hooks/transaction/useSendBatchTransaction.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/SignatureScreen.tsx`, `packages/thirdweb/src/react/core/hooks/wallets/useConnect.ts`, `packages/thirdweb/src/react/web/hooks/transaction/useSendAndConfirmTransaction.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Details.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModalContent.tsx`, `packages/thirdweb/src/react/web/hooks/transaction/useEstimateGasCost.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useConnect.ts`, `packages/thirdweb/src/react/native/hooks/transaction/useSendTransaction.tsx`, `packages/thirdweb/src/react/native/hooks/wallets/useConnect.ts`, `packages/thirdweb/src/react/core/hooks/contract/useEstimate.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useCapabilities.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useCapabilities.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useCallsStatus.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useCallsStatus.ts`, `packages/thirdweb/src/react/web/ui/components/WalletImage.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/SmartWalletConnectUI.tsx`, `packages/thirdweb/src/react/core/hooks/wallets/useCapabilities.ts`, `packages/thirdweb/src/react/web/ui/TransactionButton/index.tsx`, `packages/thirdweb/src/react/native/ui/TransactionButton/TrabsactionButton.tsx`, `packages/thirdweb/src/react/core/hooks/contract/useSendBatchTransaction.ts`, `packages/thirdweb/src/react/core/hooks/contract/useSendAndConfirmTransaction.ts`, `packages/thirdweb/src/react/core/hooks/wallets/useCallsStatus.ts`, `packages/thirdweb/src/react/core/hooks/contract/useEstimateGas.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/ConfirmationScreen.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/ConnectButton.tsx`, `packages/thirdweb/src/react/web/ui/AutoConnect/AutoConnect.tsx`, `packages/thirdweb/src/react/native/ui/AutoConnect/AutoConnect.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectEmbed.tsx`, `packages/thirdweb/src/react/core/hooks/connection/AutoConnect.tsx`, `packages/thirdweb/src/react/core/hooks/transaction/button-core.ts`, `packages/thirdweb/src/react/core/hooks/wallets/useSendCalls.ts`, `packages/thirdweb/src/react/web/hooks/wallets/useSendCalls.ts`, `packages/thirdweb/src/react/native/hooks/wallets/useSendCalls.ts`, `packages/thirdweb/src/react/core/hooks/connection/useAutoConnect.ts`, `packages/thirdweb/src/react/core/hooks/contract/useSendTransaction.ts`, `packages/thirdweb/src/exports/react.native.ts`, `packages/thirdweb/src/exports/react.ts`, `packages/thirdweb/src/react/web/hooks/transaction/useSendTransaction.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by joaquim-verges** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3312?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3312** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3304** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3304?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3300** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3300?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3298** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3298?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joaquim-verges and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/react/core/connectionManager.ts [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
+import type { AsyncStorage } from "../../utils/storage/AsyncStorage.js";
+import {
+  type ConnectionManager,
+  createConnectionManager,
+} from "../../wallets/manager/index.js";
 
-export const connectionManager = /* @__PURE__ */ createConnectionManager(
-  /* @__PURE__ */ getStorage(),
-);
+let _connectManager: ConnectionManager;
```
Comment (jnsdls) [inline packages/thirdweb/src/react/core/connectionManager.ts:7]: nit `connectionManager`

```suggestion
let connectionManager: ConnectionManager;
```

**Review by jnsdls — APPROVED** [review]
small nit, otherwise LGTM

**Comment by jnsdls** [comment]
### Merge activity

* **Jun 14, 2:59 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/js/3312) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**